.circleci/config.yml

```diff
@@ -433,55 +433,93 @@ jobs:
     executor: ndk-r21e-latest-executor
     steps:
       - checkout
-      - restore-gradle-cache
-      - assemble-core-debug
-      - assemble-ui-debug
-      - assemble-core-release
-      - assemble-ui-release
-      - assemble-module:
-          module_target: "instrumentation-tests"
-          variant: "Debug"
-          inject_token: true
-      - assemble-instrumentation-test:
-          module_target: "instrumentation-tests"
-      - assemble-module:
-          module_target: "app-tests-wrapper"
-          variant: "Debug"
-      - assemble-instrumentation-test:
-          module_target: "libnavigation-core"
-      - write-workspace
+      - when:
+          condition:
+            not:
+              matches:
+                pattern: "^add-changelog.*"
+                value: << pipeline.git.branch >>
+          steps:
+            - restore-gradle-cache
+            - assemble-core-debug
+            - assemble-ui-debug
+            - assemble-core-release
+            - assemble-ui-release
+            - assemble-module:
+                module_target: "instrumentation-tests"
+                variant: "Debug"
+                inject_token: true
+            - assemble-instrumentation-test:
+                module_target: "instrumentation-tests"
+            - assemble-module:
+                module_target: "app-tests-wrapper"
+                variant: "Debug"
+            - assemble-instrumentation-test:
+                module_target: "libnavigation-core"
+            - write-workspace
 
   unit-tests-core:
     executor: ndk-r21e-latest-executor
     steps:
-      - read-workspace
-      - unit-tests-core
+      - when:
+          condition:
+            not:
+              matches:
+                pattern: "^add-changelog.*"
+                value: << pipeline.git.branch >>
+          steps:
+            - read-workspace
+            - unit-tests-core
+      - run: exit 0
```

Comment:
Why do we need the additional `exit 0` step?

Reply:
If there is no `-run` then circle ci throws build error in the file saying at least one step is required. Since `when` is conditional it is not considered as a step that is always going to be executed. Same is done in the 3rd piece of code snippet [here](https://circleci.com/docs/2.0/configuration-reference/#logic-statement-examples)
cc @LukasPaczos

Reply:
So we also need to add it to `prepare-and-assemble` job, don't we? It's currently not there.

Reply:
There is a `-checkout` that counts for at least 1 comand; so we don't need it.

Reply:
:+1: